### ISSUE TITLE
New RoR rankings crash when asking for "top 1000 results in world" for popular events

### ISSUE THREAD

**Issue by dancing-jules** [thewca/worldcubeassociation.org#4481]
When I try to show top 1000 single results for a popular event (2x2, 3x3, pyra), I get the following error: 

![grafik](https://user-images.githubusercontent.com/33940391/63054841-f09aab80-bee4-11e9-80cd-9b09f217c955.png)

This happens on both Firefox and Chrome.
This also happens for top 1000 average results. 
This doesn't happen with less popular events (e.g. Skewb, Feet)
This doesn't happen with continental rankings.
This doesn't happen with top 1000 persons.
This doesn't happen with top 100 resuls.

**Comment by viroulep** [comment]
Some more data: it actually hits a 502, and this is from nginx's error log:
```
2019/08/14 20:54:48 [error] 1101#0: *26162 upstream prematurely closed connection while reading response header from upstream, client: ****, server: www.worldcubeassociation.org, request: "GET /results/rankings/222/single?show=1000+results HTTP/1.1", upstream: "http://unix:/tmp/unicorn.wca.sock:/results/rankings/222/single?show=1000+results", host: "www.worldcubeassociation.org"
```

Our unicorn server is setup to timeout after 30s per request, which seems to be hit:
![timeout](https://user-images.githubusercontent.com/1007485/63056163-c3033180-bee7-11e9-8362-9bcc6b08bede.png)

We are aware of some performances issues with the new page, so hopefully we'll fix that kind of issues soon!

**Comment by dancing-jules** [comment]
I said it doesn't happen with continentals, but apparently the US is big enough that with 3x3 results in North America it also happens. 

The loading in general seems to be a lot slower than on the old rankings page.

**Comment by viroulep** [comment]
We actually use the exact same SQL queries, but we've dropped caching on the way (see #4480).
It used to be "one person pay the cost of an expensive query, then everyone else can benefit from this", but until #4480 is merged it's "everyone pays the cost of an expensive query" :(

**Comment by AlbertoPdRF** [comment]
Hopefully this has been fixed with #4482 and #4483.
